service-search-nls/src/main/java/fi/nls/oskari/search/channel/OgcApiSearchChannel.java

```diff
@@ -0,0 +1,202 @@
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+
+@Oskari(OgcApiSearchChannel.ID)
+public class OgcApiSearchChannel extends SearchChannel {
+
+
+    public static final String ID = "OGC_API_SEARCH_CHANNEL";
+
+    //Local configs
+    /**
+     * One can use property search.channel.OGC_API_SEARCH_CHANNEL.apikey but if really needed apikey can go here.
+     */
+    private static String apiKey = null;
+    private static String searchAddress = "https://avoin-paikkatieto.maanmittauslaitos.fi/geographic-names/features/v1/collections/placenames/items?spelling_case_insensitive=";
+    private static String additionalPlaceTypeInfoBaseUrl = "https://beta-paikkatieto.maanmittauslaitos.fi/catalogue/v1/codelistgroups/geographic-names/codelists/";
+    private static String[] localAdditionalInfoNames = {"placeType", "municipality"};
+    /**
+     * Set true if one wants to use local jsons instead of url based one
+     */
+    private static final boolean USE_LOCAL_ADDITIONAL_INFO = true;
+    /**
+     * Add languages you have translaited here as language code
+     */
+    private static String[] languageArray = {"fin", "eng", "swe"};
+
+    private static HashMap<Integer, HashMap<String, String>> placetypeCodes;
+    private static HashMap<Integer, HashMap<String, String>> municipalityCodes;
+
+    private Logger log = LogFactory.getLogger(this.getClass());
+
+    private static String CRS = null;
+    private static final String defaultCRS = "http://www.opengis.net/def/crs/EPSG/0/3067";
+
+
+    public ChannelSearchResult doSearch(SearchCriteria criteria) {
+        ChannelSearchResult result = new ChannelSearchResult();
+
+        if (placetypeCodes == null) {
+            placetypeCodes = buildAdditionalInfo(localAdditionalInfoNames[0]);
+        }
+        if (municipalityCodes == null) {
+            municipalityCodes = buildAdditionalInfo(localAdditionalInfoNames[1]);
+        }
+
+        //ApiKey Things
+        String propertyKey = PropertyUtil.get("search.channel." + ID + ".apikey");
+        if (propertyKey != null) {
+            String encodedApiKey = IOHelper.encode64(propertyKey + ":");
+            apiKey = "Basic " + encodedApiKey;
+
+        }
+        if (apiKey == null) {
+            log.error("OGC Api key was not found. OGC api will not return results.");
+        }
+        Map<String, String> headers = new HashMap<>(5);
+        headers.put("Authorization", apiKey);
+
+        String encodedSearchTerm = "";
+        try {
+            encodedSearchTerm = URLEncoder.encode(criteria.getSearchString(), StandardCharsets.UTF_8.toString());
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
+        }
+
+        String searchQuerry = searchAddress + encodedSearchTerm + "&limit=" + criteria.getMaxResults();
+
+        CRS = PropertyUtil.get("OgcApiCrs", defaultCRS);
```

Comment:
See: https://github.com/nlsfi/oskari-server-extras/blob/develop/service-search-nls/src/main/java/fi/nls/oskari/search/NLSFIGeocodingSearchChannel.java#L54-L55

I don't think this needs to be configurable. If the service can return the coordinates in other projections you should pass the map projection from `searchCriteria.getSRS()` instead. Now this only works properly for geoportals that use that same projection.

Reply:
In NLSFIGeocodingSearchChannel.java we detect if the search was requested with SRS different from the service and transform manually on the channel. You don't need to do that here if you only want to support one projection, support can be added later when needed. But the service CRS don't need to be configurable or if you really need it to be configurable it should follow the same naming pattern as other search channel props (prefixed with channel id etc)